### PR TITLE
QA-2185: Round 2: Migrate BPM provider from GIT hash to semver (application release tag) versioning 

### PR DESCRIPTION
**Pull request by ichengchang**
Ticket: [QA-2185](https://broadworkbench.atlassian.net/browse/QA-2185)

[QA-2185]: https://broadworkbench.atlassian.net/browse/QA-2185?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- Simplify workflows
- Correct PR commit hash used for tagging provider version

### CODE REVIEW

File: .github/workflows/verify_consumer_pacts.yml [hunk 1]
```diff
@@ -203,26 +203,18 @@ jobs:
...
-          echo "env.CHECKOUT_SHA=${{ env.CHECKOUT_SHA }}       # If not empty, this reflects the git commit hash of the branch being checked out (generated by Pact Broker)"
-          echo "env.CURRENT_BRANCH=${{ env.CURRENT_BRANCH }}   # This reflects the branch being checked out if CHECKOUT_BRANCH is empty"
-          echo "env.CURRENT_SHA=${{ env.CURRENT_SHA }}         # This reflects the git commit hash of the branch being checked out if CHECKOUT_BRANCH is empty"
-          echo "env.PROVIDER_BRANCH=${{ env.PROVIDER_BRANCH }} # This reflects the provider branch for pact verification"
-          echo "env.PROVIDER_SHA=${{ env.PROVIDER_SHA }}       # This reflects the provider version for pact verification"
-          echo "env.CONSUMER_BRANCH=${{ env.CONSUMER_BRANCH }} # This reflects the consumer branch for pact verification (generated by Pact Broker)"
-          echo "env.CONSUMER_SHA=${{ env.CONSUMER_SHA }}       # This reflects the consumer version for pact verification (generated by Pact Broker)"
+          echo "env.PROVIDER_BRANCH=${{ env.PACT_PROVIDER_BRANCH }} # This reflects the provider branch to switch to for pact verification"
```
Comment (jladieu) [inline .github/workflows/verify_consumer_pacts.yml:212]: Just pattern matching, but should this be `env.PROVIDER_BRANCH`, to be consistent with the `CONSUMER_{BRANCH,VERSION}` following it, and also the use on line 207?

**Review by jladieu — COMMENTED** [review]
Little confused about

`PACT_PROVIDER_VERSION` vs. `PROVIDER_VERSION`
and
`PACT_PROVIDER_BRANCH` vs. `PROVIDER_BRANCH`

Do we need to keep both sets of vars?

File: .github/workflows/verify_consumer_pacts.yml (hunk 1 above)
Comment (ichengchang) [reply .github/workflows/verify_consumer_pacts.yml]: Changed `env.PACT_PROVIDER_BRANCH` to `env.PROVIDER_BRANCH` to be consistent with other naming.

**Comment by ichengchang** [comment]
> Little confused about
> 
> `PACT_PROVIDER_VERSION` vs. `PROVIDER_VERSION` and `PACT_PROVIDER_BRANCH` vs. `PROVIDER_BRANCH`
> 
> Do we need to keep both sets of vars?

The `PACT_...` envvars are defined in `build.gradle` and are exported to jvm system properties so we have both sets of vars to use the names in `build.gradle`.

**Review by jladieu — APPROVED** [review]

**Review by knapii-developments — APPROVED** [review]

**Pull request merged**